src/NLog/LayoutRenderers/ExceptionDataLayoutRenderer.cs

```diff
@@ -0,0 +1,135 @@
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.LayoutRenderers
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Text;
+    using NLog.Common;
+    using NLog.Config;
+    using NLog.Internal;
+
+    /// <summary>
+    /// Exception information provided through 
+    /// a call to one of the Logger.*Exception() methods.
+    /// </summary>
+    [LayoutRenderer("exceptiondata")]
+    [LayoutRenderer("exception-data")]
+    [ThreadAgnostic]
+    [ThreadSafe]
+    public class ExceptionDataLayoutRenderer : LayoutRenderer, IRawValue
+    {
+        private string _dataKey;
+        /// <summary>
+        /// Gets or sets the Data Key 
+        /// </summary>
+        [DefaultParameter]
+        public string DataKey
+        {
+            get => _dataKey;
+
+            set
+            {
+                _dataKey = SanitizeDataKey(value);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionDataLayoutRenderer" /> class.
+        /// </summary>
+        public ExceptionDataLayoutRenderer()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets whether to render innermost Exception from <see cref="Exception.GetBaseException()"/>
+        /// </summary>
+        [DefaultValue(false)]
+        public bool BaseException { get; set; }
+
+        /// <inheritdoc />
+        bool IRawValue.TryGetRawValue(LogEventInfo logEvent, out object value)
```

Comment:
Think you should remove the IRawValue-interface-logic

Reply:
done